### PR TITLE
Move criusupport to separate library from JCL

### PR DESCRIPTION
**Pull request by OussamaSaoudi**
- Moved criusupport.c to its own directory
- Added CMake and UMA build instructions to link and export criu
native functions
- Created new trace file for CRIU
- Loaded criu library in CRIUSupport.java

Signed-off-by: Oussama Saoudi <oussama.saoudi@ibm.com>

### CODE REVIEW

File: jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java [hunk 1]
```diff
@@ -29,6 +29,14 @@
  */
 public final class CRIUSupport {
 	
+	static {
+		try {
+			System.loadLibrary("j9criu29"); //$NON-NLS-1$
+		} catch (UnsatisfiedLinkError e) {
+			System.out.println("No natives for JNI tests");
```
Comment (tajila) [inline jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java:36]: I think this should say something like `Unable to load libcriu`

File: runtime/include/j9lib.h.in [hunk 2]
```diff
@@ -35,6 +35,7 @@
 #define J9_TOOLS_MIGRATION_DLL_NAME "migration"
 #define J9_BCUTIL_TEST_NATIVES_DLL_NAME "bcuwhite"
 #define J9_VERIFY_DLL_NAME "j9bcv29"
+#define J9_CRIU_DLL_NAME "j9criu29"
```
Comment (keithc-ca) [inline runtime/include/j9lib.h.in:38]: Why is that added here? I don't see any uses of `J9_CRIU_DLL_NAME`.

File: runtime/criusupport/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+add_tracegen(j9criu.tdf)
+
+j9vm_add_library(j9criu SHARED
+	OUTPUT_NAME j9criu${J9VM_VERSION_SUFFIX}
+	criusupport.c
+
+	${CMAKE_CURRENT_BINARY_DIR}/ut_j9criu.c
+)
+
+
```
Comment (keithc-ca) [inline runtime/criusupport/CMakeLists.txt:32]: Please collapse multiple blank lines to one.

**Comment by keithc-ca** [comment]
Please also correct the spelling of 'library' in your commit message.

**Comment by keithc-ca** [comment]
Please address line endings compaint:
```
15:17:00  runtime/jcl/exports.cmake:640: new blank line at EOF.
```

File: jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java [hunk 4]
```diff
@@ -29,6 +29,14 @@
  */
 public final class CRIUSupport {
 	
+	static {
+		try {
+			System.loadLibrary("j9criu29"); //$NON-NLS-1$
+		} catch (UnsatisfiedLinkError e) {
+			System.out.println("Unable to load libcriu");
```
Comment (keithc-ca) [inline jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java:36]: Perhaps this should `throw new InternalError(e);`?

Comment (keithc-ca) [reply jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java:36]: This should probably also use `AccessController.doPrivileged()`.

Comment (keithc-ca) [reply jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java]: The class may be loaded from an unprivileged context, in which case it will fail unless `doPrivileged()` is used.

Comment (OussamaSaoudi) [reply jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java]: How's it look now?

**Comment by keithc-ca** [comment]
Jenkins line endings check

File: runtime/criusupport/j9criu.tdf [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+Executable=j9criu
+DATFileName=J9TraceFormat.dat
+Submodules=j9vmutil,j9util
+
+TraceAssert=Assert_CRIU_true noEnv Overhead=1 Level=1 Assert="(P1)"
+TraceAssert=Assert_CRIU_false noEnv Overhead=1 Level=1 Assert="!(P1)"
+TraceAssert=Assert_CRIU_notNull noEnv Overhead=1 Level=1 Assert="(P1) != NULL"
+TraceAssert=Assert_CRIU_mustHaveVMAccess noEnv overhead=1 Level=1 Assert="(P1)->publicFlags & J9_PUBLIC_FLAGS_VM_ACCESS"
+TraceAssert=Assert_CRIU_mustNotHaveVMAccess noEnv overhead=1 Level=1 Assert="0 == ((P1)->publicFlags & J9_PUBLIC_FLAGS_VM_ACCESS)"
```
Comment (keithc-ca) [inline runtime/criusupport/j9criu.tdf:31]: Please spell `Overhead` consistently (with a capital 'O').

**Comment by keithc-ca** [comment]
Jenkins test sanity osx,ppc64le_linux_uma,zlinux,win,win32 jdk8

**Comment by keithc-ca** [comment]
Failure in UMA build is expected:
```
Selected VM [default] by option -Xnocompressedrefs does not exist.
```

**Review by keithc-ca — APPROVED** [review]
